broker-core/src/main/java/io/zeebe/broker/subscription/message/MessageService.java

```diff
@@ -63,6 +64,7 @@ private TypedStreamProcessor createStreamProcessors(TypedStreamEnvironment env)
     final MessageDataStore messageDataStore = new MessageDataStore();
 
     return env.newStreamProcessor()
+        .keyGenerator(new KeyGenerator(0, 1))
```

Comment:
In order to ensure unique keys, I guess that the message stream processors also need keys with offset from `KeyGenerator`.

Reply:
Not sure we need this at this point. These records are not exposed to users via topic subscriptions and internally we also do not require uniqueness currently.